transportable-udfs-api/src/main/java/com/linkedin/transport/api/udf/StdUDF5.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2018 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.transport.api.udf;
+
+import com.linkedin.transport.api.data.StdData;
+
+
+/**
+ * A Standard UDF with four input arguments.
+ *
+ * @param <I1>  the type of the first input argument
+ * @param <I2>  the type of the second input argument
+ * @param <I3>  the type of the third input argument
+ * @param <I4>  the type of the fourth input argument
+ * @param <I5>  the type of the fifth input argument
+ * @param <O>  the type of the return value of the {@link StdUDF}
+ */
+// Suppressing class parameter type parameter name and arg naming style checks since this naming convention is more
+// suitable to Standard UDFs, and the code is more readable this way.
+@SuppressWarnings({"checkstyle:classtypeparametername", "checkstyle:regexpsinglelinejava"})
+public abstract class StdUDF5<I1 extends StdData, I2 extends StdData, I3 extends StdData, I4 extends StdData,
+    I5 extends StdData, O extends StdData> extends StdUDF {
+
+  /**
+   * Returns the output of the {@link StdUDF} given the input arguments.
+   *
+   * This method is called once per input record. All UDF logic should be defined in this method.
+   *
+   * @param arg1  the first input argument
+   * @param arg2  the second input argument
+   * @param arg3  the third input argument
+   * @param arg4  the fourth input argument
+   * @param arg5  the fifth input argument
+   * @return the output of the {@link StdUDF} given the input arguments.
+   */
+  public abstract O eval(I1 arg1, I2 arg2, I3 arg3, I4 arg4, I5 arg5);
+
+  /**
+   * Returns an array of file paths to be localized at the worker nodes.
+   *
+   * The Standard UDF framework localizes the files passed through this method and provides the localized file paths to
+   * {@link StdUDF#processRequiredFiles(String[])} for further processing. Users can use the pattern "#LATEST" instead
+   * of a concrete directory name in the path as a way of selecting the directory with the most recent timestamp, and
+   * hence obtaining the most recent version of a file.
+   * Example: 'hdfs:///data/derived/dwh/prop/testMemberId/#LATEST/testMemberId.txt'
+   *
+   * The arguments passed to {@link #eval(StdData, StdData, StdData, StdData, StdData)} are passed to this method as well to
```

Comment:
Please, here and everywhere, make sure that lines breaks follow the same boundary: 120 characters.

Reply:
Is checkstyle enabled in this repo? To me the line boundary is set to 160. If you would like to promote 120-char, I suggest to enforce it directly in the checkstyle